src/main/java/ch/njol/skript/effects/base/SetEffect.java

```diff
@@ -0,0 +1,151 @@
+	 * 
+	 * @param effect The SetEffect class that you are registering.
+	 * @param property The property of the syntax.
+	 * @param makeProperty The property of the syntax used for "make %types% makeProperty".
+	 * @param type The type classinfo of the syntax. Can be plural.
+	 */
+	public static void registerMake(Class<? extends SetEffect<?>> effect, String property, String makeProperty, String type) {
+		Skript.registerEffect(effect, "set " + property + " of %" + type + "% to %boolean%",
+				"set %" + type + "%'[s] " + property + " to %boolean%",
+				"make %" + type + "% " + makeProperty,
+				"force %" + type + "% to " + makeProperty,
+				"make %" + type + "% not " + makeProperty,
+				"force %" + type + "% (to not|not to) " + makeProperty);
+	}
+
+	@Override
+	@SuppressWarnings("unchecked")
+	public boolean init(Expression<?>[] exprs, int matchedPattern, Kleenean isDelayed, ParseResult parseResult) {
+		make = matchedPattern >= 2;
+		if (exprs.length == 1) {
+			if (!make)
+				throw new SkriptAPIException("There was not two expressions in the SetEffect class '" + getClass().getName() + "' exprs: " + Arrays.toString(exprs));
+			expression = (Expression<T>) exprs[0];
+			negated = matchedPattern >= 4;
+			return true;
+		}
+		expression = (Expression<T>) exprs[matchedPattern];
+		value = (Expression<Boolean>) exprs[matchedPattern ^ 1];
+		return true;
+	}
+
+	protected final boolean isNegated() {
+		return negated;
+	}
+
+	/**
+	 * Returns the expression that was registered as the type for this SetEffect.
+	 * 
+	 * @return Expression<T> for getting the values of the expression used.
+	 */
+	protected final Expression<T> getExpression() {
+		return expression;
+	}
+
+	/**
+	 * For usage when overriding the {@link #init(Expression[], int, Kleenean, ParseResult)} method.
+	 * Set the expression to be used for the SetEffect utility class.
+	 * 
+	 * @param expression The expression that contains the data value for the SetEffect.
+	 */
+	protected final void setExpression(Expression<T> expression) {
+		this.expression = expression;
+	}
+
+	/**
+	 * Return a BiComsumer that will be used to apply the boolean value to the type.
+	 */
+	protected abstract BiConsumer<T, Boolean> apply();
+
+	/**
+	 * Return the property name of this SetEffect used for the {@link #toString(Event, boolean)} method.
+	 * 
+	 * @return String representing the name of the property registered for this syntax.
+	 */
+	protected abstract String getPropertyName();
+
+	@Override
+	protected void execute(Event event) {
+		this.event = event;
+		boolean value = make ? !negated : negated ? !this.value.getSingle(event) : this.value.getSingle(event);
+		BiConsumer<T, Boolean> consumer = apply();
+		getExpression().stream(event).forEach(expression -> consumer.accept(expression, value));
+		this.event = null;
+	}
+
+	@Override
+	public String toString(@Nullable Event event, boolean debug) {
+		if (debug || event == null)
+			return "setting " + getPropertyName();
```

Comment:
is this anymore useful? i think the regular message should always suffice (and it may contain more details)

Reply:
No, but I realized that expression could be null if the user inputs `-type` as the property. Similar with property expressions.